### PR TITLE
[synchronous-mode] Add template file for synchronous mode

### PR DESCRIPTION
**Pull request by shi-su**
**- Why I did it**
The orchagent and syncd need to have the same default synchronous mode configuration. We are trying to make the synchronous mode default with the following steps:
1. Move syncd_start.sh and syncd_init_common.sh to sonic-buildimage repo
    - [ ] add the files to sonic-buildimage (This PR)
    - [ ] remove the files form sonic-sairedis
- [ ] Enable sync mode by default in sonic-buildimage and add minigraph parser
3. Move syncd_start.sh and syncd_init_common.sh back to sonic-sairedis repo (TBD)
    - [ ] add the files to sonic-sairedis
    - [ ] remove the files form sonic-buildimage 

**- How I did it**
Move syncd_start.sh and syncd_init_common.sh to sonic-buildimage and add the files to the syncd docker when building.

**- How to verify it**
Verified with a virtual switch build

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/scripts/syncd_init_common.sh [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+fi
+
+# Use temporary view between init and apply
+CMD_ARGS+=" -u"
+
+# Set synchronous mode if it is enabled in CONFIG_DB
+SYNC_MODE=$(sonic-cfggen -d -v DEVICE_METADATA.localhost.synchronous_mode)
+if [ "$SYNC_MODE" == "enable" ]; then
+    CMD_ARGS+=" -s"
+fi
```
Comment (qiluo-msft) [inline files/scripts/syncd_init_common.sh:31]: Only this part should be moved into buildimage repo, and all others belong more into original sairedis repo.

Comment (shi-su) [reply files/scripts/syncd_init_common.sh:31]: I think probably a better way could be putting this part in syncd_start.sh and move it to buildimage repo. All other parts could remain unchanged.

Comment (shi-su) [reply files/scripts/syncd_init_common.sh:31]: I will try if adding it in syncd_start.sh looks better.

File: platform/broadcom/docker-syncd-brcm/Dockerfile.j2 [hunk 2]
```diff
@@ -27,6 +27,7 @@ RUN chmod +x /usr/bin/dsserve /usr/bin/bcmcmd
 COPY ["supervisord.conf", "/etc/supervisor/conf.d/"]
 COPY ["files/supervisor-proc-exit-listener", "/usr/bin"]
 COPY ["critical_processes", "/etc/supervisor/"]
+COPY ["files/syncd_init_common.sh", "files/syncd_start.sh", "/usr/bin/"]
```
Comment (qiluo-msft) [inline platform/broadcom/docker-syncd-brcm/Dockerfile.j2:30]: Modify per docker image seems changing to many places.
Suggest make the sync/async decision outside the container, even when there is no explicit ConfigDB field/value.

**Review by lguohan — APPROVED** [review]

**Pull request closed without merging**